plugins/rapture/nexus-coreui-plugin/src/main/resources/static/rapture/NX/coreui/controller/Assets.js

```diff
@@ -162,6 +168,49 @@ Ext.define('NX.coreui.controller.Assets', {
         assetContainer = me.getAssetContainer();
 
     assetContainer.refreshInfo(gridView.up('grid').componentModel, assetModel);
+  },
+
+  /**
+   * @protected
+   * Enable 'Delete' when user has 'delete' permission.
+   */
+  bindDeleteAssetButton: function(button) {
+    var me = this,
+        component = me.getAssetList().componentModel;
+    button.mon(
+        NX.Conditions.or(
+            NX.Conditions.isPermitted('nexus:repository-admin:*:*:delete'),
+            NX.Conditions.isPermitted('nexus:repository-admin:' + component.get('format') + ':' +
+                component.get('repositoryName') + ':delete')
```

Comment:
What about the opposite case tho: you don't have *:* but DO have perms just for this particular repo?